Intersect.Server.Core/Entities/Events/CommandProcessing.cs

```diff
@@ -312,15 +312,30 @@ Stack<CommandInstance> callStack
         )
         {
             var commonEvent = EventBase.Get(command.EventId);
-            if (commonEvent != null)
+            if (commonEvent == null)
             {
-                for (var i = 0; i < commonEvent.Pages.Count; i++)
+                return;
+            }
+
+            for (var i = 0; i < commonEvent.Pages.Count; i++)
+            {
+                if (Conditions.CanSpawnPage(commonEvent.Pages[i], player, instance))
                 {
-                    if (Conditions.CanSpawnPage(commonEvent.Pages[i], player, instance))
-                    {
-                        var commonEventStack = new CommandInstance(commonEvent.Pages[i]);
-                        callStack.Push(commonEventStack);
-                    }
+                    var commonEventStack = new CommandInstance(commonEvent.Pages[i]);
+                    callStack.Push(commonEventStack);
+                }
+            }
```

Comment:
Is there a reason this isn't a `foreach`?

Reply:
nah, it's just old code. I can update it